dev/bots/prepare_package.dart

```diff
@@ -426,6 +428,104 @@ class ArchiveCreator {
     await _unzipArchive(gitFile, workingDirectory: minGitPath);
   }
 
+  /// Downloads an archive of every package that is present in the temporary
+  /// pub-cache from pub.dev. Stores the archives in
+  /// $flutterRoot/.pub-preload-cache.
+  ///
+  /// These archives will be installed in the user-level cache on first
+  /// following flutter command that accesses the cache.
+  ///
+  /// This assumes that all packages currently in the cache are installed from
```

Comment:
Is this a safe assumption? What about users overriding `PUB_HOSTED_URL`? https://docs.flutter.dev/community/china#configuring-flutter-to-use-a-mirror-site

Reply:
Yes, this assumption is only as a precondition to the function, inside the prepare_package script.

Edited the comment to clarify.

Reply:
oh, I missed what library we were in :)